src/pal/src/include/pal/stackstring.hpp

```diff
@@ -23,17 +23,25 @@ class StackString
 
     void DeleteBuffer()
     {
-        if (m_innerBuffer != m_buffer)
+        if ( m_innerBuffer != m_buffer)
             PAL_free(m_buffer);
 
         m_buffer = NULL;
         return;
     }
 
-    void ReallocateBuffer(SIZE_T count)
+    void ReallocateBuffer(SIZE_T count, BOOL preserve)
     {
         // count is always > STACKCOUNT here.
-        T * newBuffer = (T *)PAL_malloc((count + 1) * sizeof(T));
+        // We got so far, we will allocate a little extra
+        // to prevent frequent allocations
+#if _DEBUG
+        SIZE_T count_allocated = count;
```

Comment:
I would really prefer enforcing consistency in reserving the space instead. When I see code that reserves space for some stuff but omits some other, it immediately makes me thing it is a bug.